engine/src/main/java/org/terasology/rendering/dag/nodes/BloomBlurNode.java

```diff
@@ -57,5 +63,11 @@ protected void setupConditions(Context context) {
         RenderingConfig renderingConfig = context.get(Config.class).getRendering();
         renderingConfig.subscribe(RenderingConfig.BLOOM, this);
         requiresCondition(renderingConfig::isBloom);
+        worldRenderer = context.get(WorldRenderer.class);
+    }
+
+    @Override
+    public void propertyChange(PropertyChangeEvent event) {
```

Comment:
These identical methods obviously represent a lot of code duplication. Couldn't we keep them in ConditionDependentNode and then override them only when strictly necessary?

Reply:
Yes, I was concerned about the code duplication as well. Let's do as you suggest!

Reply:
Oh wait, I just remember why I did not make that change myself.

So, here is the "default" propertyChange:
`worldRenderer.requestTaskListRefresh();`
Obviously, this requires the `worldRenderer` variable. If we store the variable in the class, one of two things happens:

- Unknown to the developer, he inherits CDN *and* creates a worldRenderer variable in his class, meaning two worldRenderer variables exist in the class. Not a big deal functionally, but I can't say I like the redundancy. This is what currently happens, btw.
- The developer is well versed with the system and knows that the `worldRenderer` variable is available to him. However, a new-comer looks at the code and goes "where did this variable come from?", or, he assumes it's a part of Node and tries to use it in non-CD Nodes, causing irritation when it doesn't work.

I'll revert back to having `worldRenderer` in the base class if you want me to, but I feel like we need a more elegant third solution for this.

Reply:
I don't think that reasoning is particularly reasonable: if you want to extend a class rather than, say, use an instance of it as a black box, through an interface, it will make sense for you to look into how the original class is written and what fields/methods it provides. 

Even if you decide to "go solo" and replicate some of the private/protected variables, given we'll be dealing with hundreds of nodes -at most-, it will be an example of poor code quality but it won't slow down the application dramatically.